docs/README.md

```diff
@@ -0,0 +1,89 @@
+# IPC Agent
+
+The IPC Agent is a process faciliting the participation of Filecoin clients like Lotus in the InterPlanetary Consensus (formerly Hierarchical Consensus).
+
+Please refer to the [IPD Agent Design](https://docs.google.com/document/d/14lkRRv6MQYnuEfp2GoGngdD8Q5YgfE38D8HTZWKgKf4) document for details on the agent.
+
+
+# IPLD Resolver
+
+The [IPLD Resolver](../ipld/resolver) is a library that IPC Agents can use to exchange data between subnets in IPLD format.
+
+## Checkpointing
+
+The most typical use case would be the propagation of checkpoints from child subnets to the parent subnet.
+
+### Checkpoint Schema
+
+One possible conceptual model of checkpointing is depicted by the following Entity Relationship diagram:
+
+![Checkpoint Schema](diagrams/checkpoint_schema.png)
+
+It shows that the Subnet Actor in the parent subnet governs the power of validators in the child subnet by proposing _Configurations_, which the child subnet is free to adopt in its _Epochs_ when the time is right, communicating back the next adopted config via _Checkpoints_.
```

Comment:
> communicating back the next adopted config via _Checkpoints_.

Note, we don't have a feedback loop yet from child subnet to parents in place yet.

Reply:
Okay no problem, just wanted to orient the reader.